### DISCUSSION TITLE
Nested ::: blocks

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#786] [Q&A]
I want to have a callout block that can be uncollapsed, e.g.
```
:::{.callout-note collapse="true"}
## Title of Note

... stuff ...
:::
```

But I would like for the block to use also :::{.column-page} to be wider than normal when it is revealed.  Is there any way to combine the two?

I would also like to be able to have an .aside inside the block if it is possible for marginal notes to be uncollapsed.

**Comment by jjallaire** [comment]
@dragonstyle Do column divs currently escape from callouts?

**Reply by dragonstyle** [reply]
You can combine the two, though the column will affect the callout when collapsed as well as when it is revealed (the collapsed callout will also be page width). For example:

```markdown
:::{.callout-note .column-page collapse="true"}
## Title of Note

... stuff ...
:::
```

The layout system as it stands today will have trouble with asides inside a callout (it will resize the callout to take the full screen width, then place the element in the margin of the callout itself. You can likely get something close to what you're looking for with something like, though the collapse behavior won't extend to the margin element. (Not that if the callout in this case were page width (e.g. using some of the page margin), the aside will appear below the callout to prevent overlapping)


```markdown
:::{.callout-tip collapse=true}
# Title of note

This is a callout that appears in the document and is very interesting.
:::

:::{.column-margin}
This is an aside that should align next to the callout.
:::
```

which renders as:
<img width="1031" alt="Screen Shot 2022-05-03 at 10 36 29 AM" src="https://user-images.githubusercontent.com/261654/166474639-e6cbbac2-e278-4546-91dd-ad55cf9fae20.png">

I'll give some thought to how we could support collapsing and revealing asides within a collapsed callout - that seems like a useful behavior (and it would be nice if the aside that was a child of the callout itself knew enough to align itself with the callout content, too!).

**Reply by dragonstyle** [reply]
Additional note- I did take a little time to improve our callout handling for margin content / asides. Now, if you place an aside or margin content inside a callout, we'll position the margin content properly and the content will follow any collapse behavior that was specified for that callout, making this functional:

```markdown
:::{.callout-tip collapse=true}
# Title of note

This is a callout that appears in the document and is very interesting.

:::{.column-margin}
This is an aside that should align next to the callout.
:::

:::

```

This should be available in Preview (v0.9 Build 348).
